### PR TITLE
Adding Jest [closes #7]

### LINKED ISSUE #7: Add Test Coverage
The project would benefit quite a bit from test coverage, enabling work on features with less ramp-up time and lower risk to existing functionality.

Here's a branch adding unit test coverage to 'main':
https://github.com/ThatsNinja/cyberpunk2077-hacking-solver/tree/add-test-coverage

### PR DESCRIPTION
**Pull request by ThatsNinja**
Adds initial support for unit testing, as well as React DOM testing. Also adds example unit tests for `bruter`.

npm Scripts:
- `test`
- `test:watch`

Packages:
- `jest`
- `@testing-library/jest-dom`
- `@testing-library/react`
- `ts-jest`

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "defaultSeverity": "error",
+    "extends": [
+        "tslint:recommended"
```
Comment (cxcorp) [inline tslint.json:4]: Could you add `tslint-config-prettier`?

**Comment by cxcorp** [comment]
While we're at it, could you try creating the file `.github/workflows/tests.yml` with the following content? Could benefit from running the tests with Github Actions on push/PR.

```
name: Tests
on:
  push:
    branches:
    - master
  pull_request:
    branches:
    - master
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - name: Install modules
      run: npm i
    - name: Run tests
      run: npm test
```

**Comment by ThatsNinja** [comment]
I was going to see if you'd want to add a vercel config for that very purpose!

I like GitHub Actions a lot, and it would be a pleasure.

Also, do you want to automate lint checking in CI and fail builds if TSLint doesn't pass?

**Comment by cxcorp** [comment]
Sure, as long as it plays nicely with prettier's default configs (speaking of which, should probably add a .prettierrc with prettier's defaults but that's for another PR)! Are there lint errors right now?

**Comment by ThatsNinja** [comment]
Added prettier, and there are 11 lint errors currently. Yeah, we can move out linting in CI to a subsequent issue.

**Comment by ThatsNinja** [comment]
Also added GitHub Actions config, as requested.

**Comment by cxcorp** [comment]
Looks good, thank you! 👍

Closes #7

**Pull request merged**